File: app/src/main/java/de/tum/in/tumcampus/models/DeviceUploadGcmToken.java
package de.tum.in.tumcampus.models;

import android.content.Context;

import java.math.BigInteger;
import java.security.SecureRandom;
import java.util.Date;

import de.tum.in.tumcampus.auxiliary.AuthenticationManager;
import de.tum.in.tumcampus.exceptions.NoPrivateKey;

public class DeviceUploadGcmToken {

    private DeviceVerification verification;
    private String token;
    private String signature;

    public DeviceUploadGcmToken(Context c, String token) throws NoPrivateKey {
        //Create some data
        this.token = token;
        this.verification = new DeviceVerification(c);

        //Sign this data for verification
        AuthenticationManager am = new AuthenticationManager(c);
        this.signature = am.sign(token);
    }

}

File: app/src/main/java/de/tum/in/tumcampusapp/api/app/model/DeviceUploadGcmToken.kt
package de.tum.`in`.tumcampusapp.api.app.model

import android.content.Context
import de.tum.`in`.tumcampusapp.api.app.AuthenticationManager
import de.tum.`in`.tumcampusapp.api.app.exception.NoPrivateKey

data class DeviceUploadGcmToken(var verification: DeviceVerification? = null,
                                var token: String = "",
                                var signature: String = "") {
    companion object {
        @Throws(NoPrivateKey::class)
        fun getDeviceUploadGcmToken(c: Context, token: String): DeviceUploadGcmToken {
            return DeviceUploadGcmToken(
                    verification = DeviceVerification.getDeviceVerification(c),
                    token = token,
                    signature = AuthenticationManager(c).sign(token)
            )
        }
    }
}
